R/l_pkconc.R

```diff
@@ -0,0 +1,196 @@
+#'   - `cero_str`: The string to use for 0 values.
+#'   - `align`: The alignment for the variable (e.g., "center").
+#'   - `format_fun`: The formatting function to use ("round" or "signif").
+#'   - `digits`: The number of digits to use for numeric formatting.
+#'
+#' @examples
+#' \dontrun{
+#'   # Example usage:
+#'   listings <- l_pkconc(adpc, listgroup_vars, grouping_vars, displaying_vars)
+#' }
+#'
+#' @import dplyr formatters rlistings
+#' @export
+#' @author Gerardo Rodriguez
+l_pkconc <- function(
+  adpc = read.csv("inst/shiny/data/DummyRO_ADNCA.csv"),
+  listgroup_vars = c("PARAM", "PCSPEC", "ROUTE"),
+  grouping_vars = c("TRT01A", "USUBJID", "AVISIT"),
+  displaying_vars = c("NFRLT", "AFRLT", "AVAL"),
+  formatting_vars_table = NULL,
+  main_title = paste0("Listing of PK Concentration by Treatment Group,",
+                      "Subject and Nominal Time, PK Population"),
+  subtitle_lists = NULL,
+  footnote_table = "*: Patients excluded from the summary table and mean plots"
+) {
+
+  # If there are columns defined in the function that are not in the data, throw an error
+  missing_cols <- setdiff(c(grouping_vars, displaying_vars), colnames(adpc))
+  if (length(missing_cols) > 0) {
+    stop(paste("Missing required columns:", paste(missing_cols, collapse = ",")))
+  }
+
+  # If the title was not user defined use the listgroup_vars argument to do it
+  if (is.null(subtitle_lists)) {
+    subtitle_lists <- paste(paste0("!", listgroup_vars),
+                            paste0("$", listgroup_vars),
+                            sep = ": ",
+                            collapse = "\n")
+  }
+
+  # If the formatting table was not user defined make one standard
+  if (is.null(formatting_vars_table)) {
+    formatting_vars_table <-  data.frame(
+      var_name = c(grouping_vars, displaying_vars)
+    ) %>%
+      rowwise() %>%
+      # Create a label column
+      mutate(
+        Label = parse_annotation(adpc, paste0("!", var_name)),
+        na_str = "NA",
+        cero_str = ifelse(var_name == "AVAL", "BLQ", "0"),
+        align = "center",
+        format_fun =  ifelse(is.numeric(adpc[[var_name]]) &&
+                               !is.integer(adpc[[var_name]]), "round", NA),
+        digits = ifelse(is.numeric(adpc[[var_name]]) &&
+                          !is.integer(adpc[[var_name]]), 3, NA)
+      ) %>%
+      # For the default case (when possible) label includes unit
+      mutate(
+        Label = if ("AVALU" %in% names(adpc)) ifelse(var_name == "AVAL",
+                                                     paste0(Label, " ($AVALU)"), Label) else Label,
+        Label = if ("RRLTU" %in% names(adpc)) ifelse(var_name == "AFRLT",
+                                                     paste0(Label, " ($RRLTU)"), Label) else Label,
+        Label = if ("RRLTU" %in% names(adpc)) ifelse(var_name == "NFRLT",
+                                                     paste0(Label, " ($RRLTU)"), Label) else Label
```

Comment:
suggestion:
1. Consider using `case_when`, should do the trick.

```suggestion
        Label = case_when(
          "AVALU" %in% names(adpc) & var_name == "AVAL" ~ paste0(Label, " ($AVALU)"),
          "RRLTU" %in% names(adpc) & var_name == "AFRLT ~ paste0(Label, " ($RRLTU)"),
          "RRLTU" %in% names(adpc) & var_name == "NFRLT ~ paste0(Label, " ($RRLTU)"),
          .default = Label
        )
```

2. Why the `$` notation? Is that the literal string you wish to have in the label? If you want to have the value/unit itself, that would probably require joining the table and referencing the appropriate column, unless this notation is parsed by `rlistings`?

Reply:
Regarding question in 2. the function is being parsed using `parse_annotation()` which is done before creating the listing object